### PR TITLE
NovAtel OEM6 API

### PR DESCRIPTION
**Pull request by CatGarab**
Adding an API for the NovAtel OEM6 GPS receiver

### CODE REVIEW

**Comment by CatGarab** [comment]
[Novatel OEM GPS](https://trello.com/c/LU7jpmgz/283-novatel-oem-gps)

File: apis/novatel-oem6-api/src/oem6.rs [hunk 1]
```diff
@@ -0,0 +1,730 @@
...
+                true => response_send
+                    .try_send((hdr, body))
+                    .or_else::<TrySendError<(Header, Vec<u8>)>, _>(|err| match err {
+                        // Our buffer is full, but the receiver should still be alive, so let's keep going
+                        TrySendError::Full(_) => Ok(()),
+                        // The receiver is gone. If both receivers are gone, then there's no point in this
+                        // loop still trying
+                        TrySendError::Disconnected(_) => {
+                            if log_err {
+                                panic!()
```
Comment (plauche) [inline apis/novatel-oem6-api/src/oem6.rs]: Do we want to embed a message in these `panic`s or log something to help with diagnosing issues?

File: apis/novatel-oem6-api/src/oem6.rs [hunk 2]
```diff
@@ -0,0 +1,730 @@
...
+        };
+
+        let conn = Arc::new(Mutex::new(Connection::from_path(bus, settings, TIMEOUT)?));
+
+        Ok(OEM6 {
+            conn,
+            log_recv,
+            response_recv,
+        })
+        //TODO: Turn off RXSTATUSEVENTA messages (UNLOG)
```
Comment (plauche) [inline apis/novatel-oem6-api/src/oem6.rs]: Found a TODO :eyes:

File: apis/novatel-oem6-api/src/oem6.rs [hunk 3]
```diff
@@ -0,0 +1,730 @@
...
+    ///
+    /// // Unlog everything. This will override the position request's `hold=true` value and
+    /// // also unlog that request
+    /// oem.request_unlog_all(true)?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// [`OEMError`]: enum.OEMError.html
+    pub fn request_unlog_all(&self, hold: bool) -> OEMResult<()> {
```
Comment (plauche) [inline apis/novatel-oem6-api/src/oem6.rs]: Can we change `hold` to something more descriptive like `clear_holds`?

Comment (CatGarab) [reply apis/novatel-oem6-api/src/oem6.rs]: That's a much better name :) all I could think of was 'override', which is a reserved keyword

File: apis/novatel-oem6-api/src/messages/commands/mod.rs [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    InvalidChannel = 20,
+    /// The requested rate is invalid
+    InvalidRate = 21,
+    /// The word has no mask for this type of log
+    NoMask = 22,
+    /// Channels are locked due to an error
+    LockedChannels = 23,
+    /// The injected time is invalid
+    InvalidTime = 24,
+    /// The COM/USB port is not supported
```
[the remark this answers is not in the document]
Comment (CatGarab) [reply apis/novatel-oem6-api/src/messages/commands/mod.rs]: Heh, true. I just copied all the possible values out of the documentation.

File: apis/novatel-oem6-api/src/messages/commands/unlog.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
+//
```
[the remark this answers is not in the document]
Comment (CatGarab) [reply apis/novatel-oem6-api/src/messages/commands/unlog.rs]: (`log`, `unlog`, and `unlog_all` are commands)

**Pull request merged**